File: ktlang/app/src/test/kotlin/ktlang/ArrayBinarySearchTest.kt
import kotlin.test.Test
import kotlin.test.assertEquals
import ktlang.array.binary.search.arrayBinarySearch

class ArrayBinarySearchTest {

    @Test
    fun findTarget() {
        assertEquals(-1, arrayBinarySearch(listOf(1, 2, 3, 4, 5, 9, 12), 8))
        assertEquals(4, arrayBinarySearch(listOf(1, 2, 3, 4, 5), 5))
        assertEquals(7, arrayBinarySearch(listOf(1, 2, 3, 5, 9, 12, 32, 41), 41))
        assertEquals(0, arrayBinarySearch(listOf(1, 2, 3, 12, 32, 5), 1))
    }
}

File: tslang/test/array/binary-search/ArrayBinarySearch.test.ts
import { expect, test } from "bun:test";
import { arrayBinarySearch } from "../../../src/array/binary-search/ArrayBinarySearch";

test("test arrayBinarySearch", () => {
	expect(arrayBinarySearch([1, 2, 3, 4, 5, 9, 12], 8)).toBe(-1);
	expect(arrayBinarySearch([1, 2, 3, 4, 5], 5)).toBe(4);
	expect(arrayBinarySearch([1, 2, 3, 5, 9, 12, 32, 41], 41)).toBe(7);
	expect(arrayBinarySearch([1, 2, 3, 12, 32, 5], 1)).toBe(0);
});
